ios/objc/apis/TorchTensor.h

```diff
@@ -0,0 +1,102 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, TorchTensorType) {
+    TorchTensorTypeByte, //8bit unsigned integer
+    TorchTensorTypeInt,  //32bit signed integer
+    TorchTensorTypeLong, //64bit signed integer
+    TorchTensorTypeFloat, //32bit single precision floating point
+    TorchTensorTypeUndefined, //Undefined tensor type. This indicates an error with the model
+};
+
+@interface TorchTensor : NSObject<NSCopying>
+/**
+ Data type of the tensor
+ */
+@property(nonatomic,assign, readonly) TorchTensorType type;
+/**
+//The size of the tensor. The returned value is a array of integer
+ */
+@property(nonatomic,strong, readonly) NSArray<NSNumber* >* size;
+/**
+ /The number of dimensions of the tensor.
+ */
+@property(nonatomic,assign, readonly) int64_t dim;
+/**
+ The total number of elements in the input tensor.
+ */
+@property(nonatomic,assign, readonly) int64_t numel;
+/**
+ Returns if the tensor has a quntized backend
+ */
+@property(nonatomic,assign, readonly) BOOL quantized;
+/**
+ Creat a tensor object with data type, shape and a raw pointer to a data buffer.
+
+ @param type Data type of the tensor
+ @param size Size of the tensor
+ @param data A raw pointer to a data buffer
+ @return  A tensor object
+ */
++ (nullable TorchTensor* )newWithType:(TorchTensorType)type Size:(NSArray<NSNumber* >*)size Data:(void* _Nullable)data;
+/**
+ Creat a tensor object with data type, shape and a raw pointer to a data buffer.
+ 
+ @param type Data type of the tensor
+ @param size Size of the tensor
+ @param data A raw pointer to a data buffer
+ @return  A tensor object
+ */
++ (nullable TorchTensor* )newWithType:(TorchTensorType)type Size:(NSArray<NSNumber* >*)size Data:(void* _Nullable)data Quantized:(BOOL) quantized;
+
+@end
+
+@interface TorchTensor(Operations)
```

Comment:
I think it'd be very useful to provide a simple way to convert a tensor to a native list of floats/ints/etc. The thing is that subscripting is pretty slow (create a tensor every time). In C++ API we provide TensorAccessor (https://pytorch.org/cppdocs/notes/tensor_basics.html#efficient-access-to-tensor-elements) or one can just do .data(). We should do something similar